src/3d/qgs3dmapsettings.cpp

```diff
@@ -128,6 +129,13 @@ void Qgs3DMapSettings::readXml( const QDomElement &elem, const QgsReadWriteConte
               elemOrigin.attribute( QStringLiteral( "y" ) ).toDouble(),
               elemOrigin.attribute( QStringLiteral( "z" ) ).toDouble() );
 
+  QDomElement elemExtent = elem.firstChildElement( QStringLiteral( "extent" ) );
+  mExtent = QgsRectangle(
+              elemExtent.attribute( QStringLiteral( "xMin" ) ).toDouble(),
+              elemExtent.attribute( QStringLiteral( "yMin" ) ).toDouble(),
+              elemExtent.attribute( QStringLiteral( "xMax" ) ).toDouble(),
+              elemExtent.attribute( QStringLiteral( "yMax" ) ).toDouble() );
```

Comment:
Maybe we could use QgsXmlUtils and its readRectangle and writeRectangle methods?

Reply:
I chose not to use the `QgsXmlUtils::writeRectangle()` method as it writes the extrema values as separate elements, while the general paradigm for 3d settings is to store values as attributes.